### ISSUE TITLE
Is it can export the result with tree view?

### ISSUE THREAD

**Issue by msaong** [reworkd/AgentGPT#82]
The tool is very powerful, but the amount of information generated after the program finishes execution is huge. The current export methods are difficult to organize. Is it possible to use a tree structure to organize the results by levels, for example, exporting in formats such as markdown or JSON?

**Comment by awtkns** [comment]
This is something we are looking to add in the future. 😊

We also want to display the info better in the ui.

**Comment by psgit** [comment]
great idea 👍 

I think that an initial implementation of a tree in JSON could be done relatively easily from the existing message structure. Below is an initial suggestion, with an example :

```json
{
    "agent": "find the weather",
    "goal": "find the weather in berlin for today",
    "tasks": [{
            "name": "Retrieve current date and time",
            "executions": [{
                "response": "The current date and time is [insert current date and time for Berlin here].",
                "ttasks": []
            }]
        },
        {
            "name": "Retrieve current weather conditions in Berlin",
            "executions": [{
                "response": "The current weather in Berlin is mostly cloudy with a temperature of 11°C and a chance of rain later in the day.",
                "tasks": []
            }]
        }
    ]
}
```

**Comment by psgit** [comment]
I think that an initial implentation in Markdown, with a similar structure, could also be done relatively easily.

**Comment by asim-shrestha** [comment]
@psgit Is this something you'd be able to work on?

**Comment by psgit** [comment]
@asim-shrestha yes, I would indeed be able to work on it. You can assign the issue to me, if you want. 👍

**Comment by psgit** [comment]
@asim-shrestha  I have created a simple tree view with the messages in the chat window, below an example screenshot: 
![Example simple tree view](https://user-images.githubusercontent.com/7589105/233559578-17f186a7-3287-44b3-9868-dfaf893b3335.jpg)

A markdown file, with embedded JSON, can also be saved with the Treeview button, in the header. I have uploaded an [example](https://github.com/psgit/AgentGPT/blob/feat/task-82-tree-view-ui/example-agent-treeview.md).

These still need a lot more fine-tuning and debugging.

**Comment by asim-shrestha** [comment]
Awesome @psgit 

Would love to get this into the actual project, though style wise we may want to keep it true to what it was originally. However, we can update the `TaskWindow` to adopt this approach. 

Let us know if you're still interested!

**Comment by psgit** [comment]
@asim-shrestha I am indeed still interested. I have merged the main branch into the feature branch to get the latest changes to AgentGPT and am testing with the updated branch.

I am open to any suggestions regarding style.

**Comment by jasangill1** [comment]
@psgit Hello,

Are you still working on this ?? If so for the styling could you go with a theme that is like our current application this would make it easier to implement and merge!! 

Please let me know if this is something you can do, otherwise, I will be closing due to inactivity!!
